param/hash.go

```diff
@@ -0,0 +1,7 @@
+package param
```

Comment:
The name is perhaps not good. The idea is to have one package that can provide system-wide interfaces and consts.

Reply:
This approach increases coupling, what would be good to avoid.

Reply:
We should discuss this, because it's not clear to me where the interface actually belongs. The way it is now it can only be in `bmt`, not because it most belongs there, but because it's the only package relevant to it that doesn't create circular dependency.

Would you accept a sub-package of file having this interface?

Reply:
I would organise like this. Package`/file` should contain SectionWriter interface and all types and variables that would be common to all implementations. All implementations should be in `/file` subdirectories as packages with appropriate names.